### ISSUE TITLE
Error:  newtonsoft.json library it referenced did not contain a reference to monotouch.droid. 

### ISSUE THREAD

**Issue by mhdmaktabi** [step-up-labs/firebase-database-dotnet#155]
Xamarin project made a .net standard(1.4) shared project and tried referencing this library from it. I tried making a build which said that the newtonsoft.json library it referenced did not contain a reference to monotouch.droid. 

Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'Newtonsoft.Json, Version=8.0.0.0, Culture=neutral, ....'. Perhaps it doesn't exist in the Mono for Android profile?
File name: 'Newtonsoft.Json.dll'
   at Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve(AssemblyNameReference reference, ReaderParameters parameters)
   at Xamarin.Android.Tasks.ResolveAssemblies.AddAssemblyReferences(DirectoryAssemblyResolver resolver, ICollection`1 assemblies, AssemblyDefinition assembly, Boolean topLevel)
   at Xamarin.Android.Tasks.ResolveAssemblies.Execute(DirectoryAssemblyResolver resolver)	SmartHotel.Clients.Android

**Comment by mhdmaktabi** [comment]
@cabauman Do you have any idea?

**Comment by cabauman** [comment]
Sorry, not sure. I'd have to see the project and look around.

**Comment by mhdmaktabi** [comment]
Could I provide any further details?

**Comment by mhdmaktabi** [comment]
@bezysoftware  Do you have any idea?

**Comment by cabauman** [comment]
@mhdmaktabi If you link me to your project, I'll check it out. Also, I made a sample Xamarin.Forms project if you want to compare: https://github.com/cabauman/firebase-database-dotnet/tree/xamarin_forms_sample/samples/XamarinForms

**Comment by cabauman** [comment]
Do you have System.Reactive 4.0.0 installed?

**Comment by mhdmaktabi** [comment]
@cabauman Yes I have System.Reactive 4.0.0 installed.

**Comment by cabauman** [comment]
Okay, that's the problem. This library is targeting System.Reactive 3.1.1. So you have two options:

1) Downgrade System.Reactive to 3.1.1
2) Install System.Reactive.Compatibility

**Comment by bezysoftware** [comment]
I tried to upgrade the Rx extensions to v4, but this broke the .net native UWP pipeline in my other project (it simply froze). Until this is resolved it will need to stay on 3.1.1

**Comment by mhdmaktabi** [comment]
Thank you @cabauman this worked for me.
@bezysoftware Yes exactly!

**Comment by cabauman** [comment]
@bezysoftware Hmm okay. I’m pretty active in the Rx.Net slack channel. Haven’t seen any issues. I’ll ask around. Anymore details on when it froze?

**Comment by bezysoftware** [comment]
@cabauman thanks, I don't have a repro project yet, that's why I haven't posted it on the rx github yet. It must be related to either System.Reactive (3.1.1 x 4.1.0) or ReactiveUI (8.0.0 x 8.7.2). With the newer versions the build gets stuck here at "Running 'Analysis' transform, heap at 14 MB."

```
2018-09-02T12:31:53.3806135Z BuildNativePackage:
2018-09-02T12:31:53.3808479Z   Creating ilc response file "obj\x86\Release\ilc.SettleUp.rsp" with the following content: /in:"C:\agent\_work\5\s\SettleUp\obj\x86\Release\ilc\in" /out:"C:\agent\_work\5\s\SettleUp\bin\x86\Release\ilc" /intermediatespath:"C:\agent\_work\5\s\SettleUp\obj\x86\Release\ilc\intermediate" /v:normal /keepintermediates:true /buildtype:ret /exename:"SettleUp.exe" /rcpath:"C:\Program Files %28x86%29\Windows Kits\10\bin\10.0.16299.0\x64\rc.exe" /targetplatformsdklibpath:"C:\Program Files %28x86%29\Windows Kits\10\lib\10.0.16299.0\um" /WindowsSdkApiXml:"C:\Program Files %28x86%29\Windows Kits\10\\App Certification Kit\SupportedAPIs-x86.xml" /targetplatformsdkmetadatapath:"C:\agent\_work\5\s\SettleUp\obj\x86\Release\ilc\in\WinMetadata" /RdXmlPath:"C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\microsoft.net.native.compiler\2.1.8\tools\LibraryXML" /CscPath:"C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\microsoft.net.native.compiler\2.1.8\tools\csc\csc.exe" /suppressPDBWarnings:true /AllowNoManagedCode /frameworkDependency:"Name = Microsoft.NET.Native.Framework.2.1, MinVersion = 2.1.26424.0, Publisher = 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'" /frameworkDependency:"Name = Microsoft.NET.Native.Runtime.2.1, MinVersion = 2.1.26424.0, Publisher = 'CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US'"  /UseSharedAssemblies /InSharedAssemblyPath:"C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\runtime.win10-x86.microsoft.net.native.sharedlibrary\2.1.8\build\..\tools\SharedLibrary"
2018-09-02T12:31:53.3893232Z   C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\runtime.win10-x86.microsoft.net.native.compiler\2.1.8\build\..\tools\\x86\ilc\ilc.exe @"obj\x86\Release\ilc.SettleUp.rsp" "/logger:CsvLogger;4920;5060" 
2018-09-02T12:31:53.9455281Z   Launching 'C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\runtime.win10-x86.microsoft.net.native.compiler\2.1.8\tools\x86\ilc\ilc.exe --gatekeeper @"C:\agent\_work\5\s\SettleUp\obj\x86\Release\ilc\intermediate\gkargs.rsp"'
2018-09-02T12:31:55.4063928Z   'C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\runtime.win10-x86.microsoft.net.native.compiler\2.1.8\tools\x86\ilc\ilc.exe --gatekeeper @"C:\agent\_work\5\s\SettleUp\obj\x86\Release\ilc\intermediate\gkargs.rsp"' completed.
2018-09-02T12:31:55.4065771Z   Process execution statistics (all times in milliseconds)
2018-09-02T12:31:55.4066362Z   -----------------------------------------------------------------------------
2018-09-02T12:31:55.4067588Z   launch  |  start   |   stop   |   end    | duration | lifetime | [slot] name
2018-09-02T12:31:55.4068369Z   -----------------------------------------------------------------------------
2018-09-02T12:31:55.4069203Z          0 |      157 |     1313 |     1454 |     1156 |     1454 | [0] Gatekeeper
2018-09-02T12:31:55.4069800Z   Number of commands:      1
2018-09-02T12:31:55.4070324Z   Total system time:       1454 ms
2018-09-02T12:31:55.4071004Z   System time per command: 1454 ms
2018-09-02T12:31:55.4071743Z   Total duration:          1156 ms
2018-09-02T12:31:55.4072189Z   Total lifetime:          1454 ms
2018-09-02T12:31:55.4072578Z   Requested DOP:           2
2018-09-02T12:31:55.4072995Z   Duration DOP:            0.8
2018-09-02T12:31:55.4073556Z   Lifetime DOP:            1.0
2018-09-02T12:31:56.1867598Z   Processing application code
2018-09-02T12:31:56.2059926Z   Running 'DumpNativeResources' transform, heap at 4 MB.
2018-09-02T12:31:56.3153423Z   Running 'Analysis' transform, heap at 14 MB.
2018-09-02T13:29:42.8725391Z Attempting to cancel the build...
2018-09-02T13:29:43.8896751Z ##[warning]C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\microsoft.net.native.compiler\2.1.8\tools\Microsoft.NetNative.targets(788,5): Warning MSB5021: "ilc" and its child processes are being terminated in order to cancel the build.
2018-09-02T13:29:43.8899624Z C:\Windows\ServiceProfiles\NetworkService\.nuget\packages\microsoft.net.native.compiler\2.1.8\tools\Microsoft.NetNative.targets(788,5): warning MSB5021: "ilc" and its child processes are being terminated in order to cancel the build. [C:\agent\_work\5\s\SettleUp\SettleUp.csproj]
2018-09-02T13:29:44.3317927Z Done Building Project "C:\agent\_work\5\s\SettleUp.sln" (default targets) -- FAILED.
2018-09-02T13:29:44.4194118Z Done Building Project "C:\agent\_work\5\s\SettleUp\SettleUp.csproj" (default targets) -- FAILED.
```

**Comment by cabauman** [comment]
Ah so you use ReactiveUI too. Cool. That’s actually in the same slack channel as Rx.Net. I mostly work with Xamarin but I have a native UWP project in the works too.

**Comment by bezysoftware** [comment]
This should be fixed with vs 2017 15.9 https://developercommunity.visualstudio.com/content/problem/339371/uwp-app-release-build-with-systemreactive-410-and.html
I will update to 4.1 then

**Comment by cabauman** [comment]
Great news.
